### PR TITLE
Port ajax-produced strings on dashboards

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Translates the budget bar courtesy of some great work by @DarthHater in #1597 . Also translates the `render_async` stuff in the budget bar.

This leaves the tooltips alone, as they are a LOT of spanish. But that's the remaining part.

This pull request makes the following changes:
* Translates search results partial and budget bar

It relates to the following issue #s: 
* Bumps #1575

### CODE REVIEW

File: config/locales/es.yml [hunk 1]
```diff
@@ -20,8 +20,12 @@ es:
     activity_log:
       label: Actividad de línea de llamadas %{current_line}
     budget_bar:
-      sent: "%{amount} expedido (%{count} pacientes)"
```
Comment (montanezp) [inline config/locales/es.yml]: "%{amount} enviado (%{count} pacientes)"

File: config/locales/es.yml [hunk 2]
```diff
@@ -20,8 +20,12 @@ es:
...
-      pledged: "%{amount} prometido (%{count} pacientes)"
+      pop_content_appt_date: cita en %{date}
+      pop_content_no_appt_date: sin cita
+      sent_item: "%{amount} expedido"
```
Comment (montanezp) [inline config/locales/es.yml]: "%{amount} enviado"

File: config/locales/es.yml [hunk 3]
```diff
@@ -20,8 +20,12 @@ es:
...
+      pledged_item: "%{amount} prometido"
+      sent_report: "%{amount} expedido (%{count} pacientes)"
```
Comment (montanezp) [inline config/locales/es.yml]: "%{amount} enviado (%{count} pacientes)"

File: config/locales/es.yml [hunk 4]
```diff
@@ -19,6 +19,13 @@ es:
   dashboard:
     activity_log:
       label: Actividad de línea de llamadas %{current_line}
+    budget_bar:
+      pledged_item: "%{amount} prometido"
+      pledged_report: "%{amount} prometido (%{count} pacientes)"
+      pop_content_appt_date: cita en %{date}
+      pop_content_no_appt_date: sin cita
+      sent_item: "%{amount} expedido"
```
Comment (montanezp) [inline config/locales/es.yml]: "%{amount} enviado"

File: app/controllers/dashboards_controller.rb [hunk 5]
```diff
@@ -22,6 +22,10 @@ def search
   end
 
   def budget_bar
+    # i18n-tasks-use t('dashboard.budget_bar.pledged_item')
+    # i18n-tasks-use t('dashboard.budget_bar.sent_item')
+    # i18n-tasks-use t('dashboard.budget_bar.pledged_report')
+    # i18n-tasks-use t('dashboard.budget_bar.sent_report')
```
Comment (lomky) [inline app/controllers/dashboards_controller.rb]: Do we wanna clean these up, or are they place-holding for a soon-to-come thing?

File: app/views/dashboards/_budget_bar.html.erb [hunk 6]
```diff
@@ -15,13 +19,18 @@
...
+                                           unit: '$',
+                                           format: '%u%n'),
+                count: expenditures[type].count) %>
         </div>
       <% end %>
       <%# TODO Commenting out until the config is ready %>
       <!-- 
       <div class="col-sm-4">
-        <%= number_to_currency(budget_bar_remaining(expenditures, limit), precision: 0) %> remaining
+        <%#= number_to_currency(budget_bar_remaining(expenditures, limit), precision: 0) %> remaining
```
Comment (lomky) [inline app/views/dashboards/_budget_bar.html.erb]: Double commenting out: when you for reals don't want it running! :grinning:

File: app/views/dashboards/_search_form.html.erb [hunk 7]
```diff
@@ -4,6 +4,8 @@
   </h1>
 
   <%= bootstrap_form_tag url: '/search', remote: true, layout: :inline do |p| %>
+    <%= p.hidden_field :locale, value: params[:locale] %>
```
Comment (lomky) [inline app/views/dashboards/_search_form.html.erb]: :rocket:

File: app/controllers/dashboards_controller.rb (hunk 5 above)
Comment (colinxfleming) [reply app/controllers/dashboards_controller.rb]: ah, thanks for calling this out -- we call on these via interpolation, so i18n health thinks they aren't being used. By declaring them in the controller we're sending a `noqa` type message to the health checker, letting it know that ACTUALLY we're using these, just indirectly. I'll add a comment.

**Pull request merged**